File: Queue/Java/Queue.java
/**
 * Implementation of a typical Queue using 
 * a double linked list.
 *
 * @author Thomas Lang, 2014
 * @version 0.1
 */
public class Queue<T extends Comparable<T>>{

    DoubleLinkedList<T> q;

    /**
     * Constructor, initializes the list.
     */
    public Queue(){ q = new DoubleLinkedList<T>(); }

    /**
     * Checks whether the Queue is empty or not.
     *
     * @return true if it is empty, false otherwise
     */
    public boolean isEmpty(){ return q.isEmpty(); }

    /**
     * Enqueues a new Node containing <code>element</code> 
     * to the Queue.
     *
     * @param  T element
     *                   Content of the Node to insert
     */
    public void enqueue( T element ){
        q.addFront( element );
    }

    /**
     * Dequeues (removing the top element of the Queue)
     *
     * @return The popped element.
     */
    public Node<T> dequeue(){
        return q.pop();
    }

    /**
     * String representation of a Queue.
     * 
     * @return String representation of a Queue.
     */
    public String toString(){
        return q.toString();
    }
}

File: Queue/Python/queue.py
#!/usr/bin/env python

from double_linked_list import *

class Queue:
    """Class representing a Queue which is
       simply a double linked list. """

    def __init__( self ):
        """Creates a new but empty Queue."""
        self.dlist = List()


    def isEmpty( self ):
        """Checks wheter the Queue is empty or not.
        
        @return true if the Queue is empty,
                false otherwise.  
        """
        return self.dlist.isEmpty()


    def enqueue( self, value ):
        """Adds a new node containing "value" 
        at the front of the Queue.
        
        @param  value  The new value to add."""
        node = Node( value )
        self.dlist.addFront( node )


    def dequeue( self ):
        """Deletes the last element of the Queue."""
        self.dlist.pop()


    def __str__( self ):
        """Returns a String representation of the Queue.
        
        @return A String representation of the Queue.
        """
        return self.dlist.__str__()


def main():
    print( "*** Creating new Queue ... done." )
    q = Queue()

    print( "*** Enqueueing values 0, 7, 42, 999 ... done." )
    q.enqueue( 0   )
    q.enqueue( 7   )
    q.enqueue( 42  )
    q.enqueue( 999 )

    print( "   ### Current Queue: " )
    print( q )

    print( "*** Dequeueing ... done." )
    q.dequeue()

    print( "   ### Current Queue: " )
    print( q )



if __name__ == '__main__':
    main()

